src/components/fields/select-field/select-field.js

```diff
@@ -0,0 +1,176 @@
+    value: (props, ...rest) =>
+      props.isMulti
+        ? PropTypes.arrayOf(PropTypes.string).isRequired(props, ...rest)
+        : PropTypes.string(props, ...rest),
+    onChange: PropTypes.func.isRequired,
+    onBlur: PropTypes.func,
+    maxMenuHeight: PropTypes.number,
+    isDisabled: PropTypes.bool,
+    isMulti: PropTypes.bool,
+    components: PropTypes.object,
+    options: PropTypes.arrayOf(
+      PropTypes.oneOfType([
+        PropTypes.shape({ value: PropTypes.string.isRequired }),
+        PropTypes.shape({
+          options: PropTypes.arrayOf(
+            PropTypes.shape({ value: PropTypes.string.isRequired })
+          ),
+        }),
+      ])
+    ),
+    noOptionsMessage: PropTypes.func,
+
+    // LabelField
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+    hint: requiredIf(
+      PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+      props => props.hintIcon
+    ),
+    description: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    onInfoButtonClick: PropTypes.func,
+    hintIcon: PropTypes.node,
+    badge: PropTypes.node,
+  };
+
+  static defaultProps = {
+    horizontalConstraint: 'scale',
+  };
+
+  state = {
+    // We generate an id in case no id is provided by the parent to attach the
+    // label to the input component.
+    id: this.props.id,
+  };
+
+  static getDerivedStateFromProps = (props, state) => ({
+    id: do {
+      if (props.id) props.id;
+      else if (state.id) state.id;
+      else sequentialId();
+    },
+  });
+
+  render() {
+    const hasError =
+      SelectInput.isTouched(this.props.touched) && hasErrors(this.props.errors);
+    return (
+      <Constraints.Horizontal constraint={this.props.horizontalConstraint}>
+        <Spacings.Stack scale="xs">
+          <FieldLabel
+            title={this.props.title}
+            hint={this.props.hint}
+            description={this.props.description}
+            onInfoButtonClick={this.props.onInfoButtonClick}
+            hintIcon={this.props.hintIcon}
+            badge={this.props.badge}
+            hasRequiredIndicator={this.props.isRequired}
+            htmlFor={this.state.id}
+          />
+          <SelectInput
+            {...omit(this.props, [
+              // We need to forward any props to SelectInput, which forwards
+              // them to react-select.
+              // So we filter out all props SelectField uses itself, so that
+              // only the props for SelectInput are forwarded.
+              'id',
+              'horizontalConstraint',
+              'hasError',
+              'hasWarning',
+              'title',
+              'hint',
+              'description',
+              'onInfoButtonClick',
+              'hintIcon',
+              'badge',
+              'hasRequiredIndicator',
+              'htmlFor',
+              'errors',
+              'isVisible',
+              'renderError',
+              // data attributes are forwarded
```

Comment:
`...omit(this.props, Object.keys(SelectField.propTypes)`?

Reply:
in this case doing something like `const { hintIcon, badge, ..., ...rest } = this.props` would have been more readable, but against our standards I suppose.

Reply:
That's pretty clever! We need to omit additional props as well like `hasError` (as we do not want to support it).

To me the better approach (compared to what we're currently doing) is to be strict about the props supported in `SelectInput` and to remove the spread there. Then we can also be explicit about the forwarded props here.

I think accepting any props is a bad approach as the API is not clearly defined and prop-spreading is generally hard to deal with.

I'd do this in a follow-up PR then though (I added it to my list).

Reply:
> in this case doing something like `const { hintIcon, badge, ..., ...rest } = this.props` would have been more readable, but against our standards I suppose.

I like the explicitness of `omit` over "abusing" destructuring for omitting purposes, especially as we might need to declare some variables which we then won't use (like `hasError`).

Reply:
I now created #119 which explicitly declares all props accepted by `SelectInput`. I will stop the spread in `SelectField` (here) and forward the explicitly as well.

Then we don't need this ugly omission anymore.